src/v2/System/Router/buildServerApp.tsx

```diff
@@ -255,7 +255,12 @@ export function buildServerApp(
         // extractor.getLinkTags()
 
         // Get serializable Relay data for rehydration on the client
-        const _relayData = await (relayEnvironment.relaySSRMiddleware as RelayServerSSR).getCache()
+        let _relayData = [] as any[]
+        try {
+          _relayData = await (relayEnvironment.relaySSRMiddleware as RelayServerSSR).getCache()
+        } catch (error) {
+          console.error(error)
```

Comment:
So what was happening is that:

404 errors within the routing framework would hit `RenderStatus => RenderError`, and since this was handled it'd be rendered normally, so on the server-side that means `buildServerApp`. But since we 404'd I guess there's nothing in the cache so this `getCache` call was throwing.

When this threw, that meant that ultimately the error would get propagated to the `errorHandlerMiddleware`, and since initially there was a 404 error which started this process, that status code was getting set correctly.

By catching this error here the `ErrorPage` renders normally within the `AppShell` — BUT we're not setting the status code anywhere so it's rendering with a `200`.